src/UKHO.S100PermitService.Common/Services/PermitService.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Microsoft.Extensions.Logging;
+using UKHO.S100PermitService.Common.Events;
+using UKHO.S100PermitService.Common.Models;
+using UKHO.S100PermitService.Common.Utilities;
+
+namespace UKHO.S100PermitService.Common.Services
+{
+    public class PermitService : IPermitService
+    {
+        private const string DateFormat = "yyyy-MM-ddzzz";
+
+        private readonly ILogger<PermitService> _logger;
+        private readonly IPermitReaderWriter _permitReaderWriter;
+
+        public PermitService(IPermitReaderWriter permitReaderWriter,
+                                ILogger<PermitService> logger) 
+        {
+            _permitReaderWriter = permitReaderWriter;
+            _logger = logger;
+        }
+
+        public void CreatePermit(DateTimeOffset issueDate, string dataServerIdentifier, string dataServerName, string userPermit,decimal version, List<Products> products)
+        {
```

Comment:
This logic should really be in IPermitService. Controllers are really orchestrators, the business logic should be implemented elsewhere.

Reply:
As discussed this is already taken care and was commented by mistake, so marking it as resolved.